### PR TITLE
Add support for modifier within the Jaxb XMLElement annotation

### LINKED ISSUE #4432: Support required modifier in the Jaxb XMLElement annotation
##### Swagger-codegen version

Discovered in:
2.2.1

Also missing in:
2.2.2-SNAPSHOT

##### Swagger declaration file content or url

    ```yaml
    definitions:
  GetEncounterByEncounterIdListRequest:
    type: object
    properties:
      cpi:
        type: string
    required:
    - cpi
    ```

##### Command line used for generation

```swagger-codegen generate -l jaxrs-cxf -Dmodels -i openapi.yaml```

I am also using the maven plugin:

```xml
<plugin>
                <groupId>io.swagger</groupId>
                <artifactId>swagger-codegen-maven-plugin</artifactId>
                <version>2.2.2-SNAPSHOT</version>
                <executions>
                    <execution>
                        <goals>
                            <goal>generate</goal>
                        </goals>
                        <configuration>
                            <inputSpec>openapi.yaml</inputSpec>
                            <language>jaxrs-cxf</language>
                            <environmentVariables>
                                <models></models>
                            </environmentVariables>
                            <configOptions>
                                <dateLibrary>joda</dateLibrary>
                            </configOptions>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
```

##### Steps to reproduce

1. Add a required field to a model declaration
2. Generate the JAXB models

##### Related issues

Not that I could find, although it appears between 2.2.1 and 2.2.2-SNAPSHOT the CXF JAXB mustache templates have changed drastically.

##### Suggest a Fix

PR Submitted

### PR DESCRIPTION
**Pull request by danmikita**
Updated README with JaxbAnnotations configuration option.

### PR checklist

- [x] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

Add support for modifier within the Jaxb XMLElement annotation.

Fixes #4432

### CODE REVIEW

**Comment by danmikita** [comment]
I'm not sure why the Travis build failed. I don't see any output when I click the link - all tests pass locally.

File: modules/swagger-codegen-maven-plugin/README.md [hunk 1]
```diff
@@ -45,7 +45,8 @@ mvn clean compile
 - `modelPackage` - the package to use for generated model objects/classes
 - `apiPackage` - the package to use for generated api objects/classes
 - `invokerPackage` - the package to use for the generated invoker objects
-- `modelNamePrefix` and `modelNameSuffix` - Sets the pre- or suffix for model classes and enums.
+- `modelNamePrefix` and `modelNameSuffix` - Sets the pre- or suffix for model classes and enums
+- `useJaxbAnnotations` - enable Jaxb annotations inside the generated models
```
Comment (wing328) [inline modules/swagger-codegen-maven-plugin/README.md]: If I understand correctly, this feature is for `jaxrs-cxf` only so I suggest we document this feature is for `jaxrs-cxf` only for time being before we add similar feature for other JAX-RS generators.

(personally I prefer `enableJaxbAnnotations` but `useJaxbAnnotations` is also fine with me)

Comment (danmikita) [inline modules/swagger-codegen-maven-plugin/README.md]: @wing328 That is correct. This feature is only for `jaxrs-cxf`. I did not add the maven extension myself, but I am more than happy to change the vernacular to `enableJaxbAnnotations`. Is that something you would like me to do to get this PR merged?

Comment (wing328) [inline modules/swagger-codegen-maven-plugin/README.md]: @danmikita I found 2 other options starting with `use` and no option starts with `enable` at the moment so I'm ok with the name you pick. 

I'll perform a bit more tests later today and merge into master if no question/feedback from anyone.

**Comment by wing328** [comment]
@danmikita thanks for the PR, which has been merged into master.

**Pull request merged**